app/models/git_repository.rb

```diff
@@ -9,6 +9,11 @@ class GitRepository
     Rails.application.config.samson.cached_repos_dir
   end
 
+  def self.valid_ref_format?(reference)
+    IO.popen(['git', 'check-ref-format', "refs/head/#{reference}"]) { |_| }
```

Comment:
No, `check-ref-format` just confirms that the reference is valid.

I'm checking for the validity of the reference before I make the call to GitHub, because if you pass a reference like `current_account['url']` into the Octokit library, it fails and throws an `URI::InvalidURIError`.

Another way of fixing this bug would be to add `rescue URI::InvalidURIError` clauses in the same places where we're currently rescuing Octokit errors.